internal/store/store.go

```diff
@@ -126,6 +126,34 @@ func (bs *BlockStore) LoadBaseMeta() *types.BlockMeta {
 	return nil
 }
 
+// LoadBlock returns the block with the given height in protobuf.
+// If no block is found for that height, it returns nil.
+func (bs *BlockStore) LoadBlockProto(height int64) *tmproto.Block {
+	var blockMeta = bs.LoadBlockMeta(height)
+	if blockMeta == nil {
+		return nil
+	}
+
+	pbb := new(tmproto.Block)
+	buf := []byte{}
+	for i := 0; i < int(blockMeta.BlockID.PartSetHeader.Total); i++ {
+		part := bs.LoadBlockPart(height, i)
+		// If the part is missing (e.g. since it has been deleted after we
+		// loaded the block meta) we consider the whole block to be missing.
+		if part == nil {
+			return nil
+		}
+		buf = append(buf, part.Bytes...)
+	}
+	err := proto.Unmarshal(buf, pbb)
+	if err != nil {
+		// NOTE: The existence of meta should imply the existence of the
+		// block. So, make sure meta is only saved after blocks are saved.
+		panic(fmt.Errorf("error reading block: %w", err))
+	}
+	return pbb
+}
+
 // LoadBlock returns the block with the given height.
 // If no block is found for that height, it returns nil.
 func (bs *BlockStore) LoadBlock(height int64) *types.Block {
```

Comment:
Could we refactor this method to use the new one, so that we don't duplicate code?